frontend/app/components/input/markdown-toolbar.tsx

```diff
@@ -22,8 +34,21 @@ const codeLabel = 'Insert a code';
 const linkLabel = 'Add a link <cmd-k>';
 const unorderedListLabel = 'Add a bulleted list';
 const orderedListLabel = 'Add a numbered list';
+const attachImageLabel = 'Attach the image, drag & drop or paste from clipboard';
 
 export default class MarkdownToolbar extends Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.uploadImages = this.uploadImages.bind(this);
+  }
+  uploadImages(e: Event) {
+    const currentTarget = (e as FileInputEvent).currentTarget;
+    if (currentTarget && currentTarget.files && currentTarget.files.length > 0 && this.props.allowUpload) {
```

Comment:
It's maybe my optinion but early return should be better
```js
  async uploadImages(e: Event) {
    const currentTarget = (e as FileInputEvent).currentTarget;
    if (!this.props.allowUpload || !currentTarget || !currentTarget.files || currentTarget.files.length === 0) return;

    const files = currentTarget.files;
    this.props.uploadImages(files);
    currentTarget.value = null;
  }
```

Also, `event.preventDefault()` would be good here, just in case

Reply:
hmm, found out, that if you're trying to paste more that one image, then pasting doesn't work because `clipboardData.files.length === 0` in that case. Even on github. Strange :-)

Reply:
You are clearing currentTarget.value (why, btw?) before upload have finished, which causes only first file to be uploaded. You should `await` there

```js
  async uploadImages(e: Event) {
    const currentTarget = (e as FileInputEvent).currentTarget;
    if (!this.props.allowUpload || !currentTarget || !currentTarget.files || currentTarget.files.length === 0) return;

    const files = currentTarget.files;
    await this.props.uploadImages(files);
    currentTarget.value = null;
  }
```

Reply:
I also prefer early return, but for this case three negative statement to read more hard than three direct positive statement.